akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -154,7 +148,40 @@ object Unmarshaller
         s"Right failure: ${right.getMessage}, " +
         s"Left failure: ${left.getMessage}")
 
+  /**
+   * Signals that unmarshalling failed because the entity content-type did not match one of the supported ranges.
+   * This error cannot be thrown by custom code, you need to use the `forContentTypes` modifier on a base
+   * [[akka.http.scaladsl.unmarshalling.Unmarshaller]] instead.
+   */
+  final case class UnsupportedContentTypeException(
+    supported:   Set[ContentTypeRange],
+    contentType: Option[ContentType]   = None)
+    extends RuntimeException(supported.mkString(
+      s"Unsupported Content-Type [$contentType], supported: ", ", ", "")) {
+
+    //constructors and copy methods added to cover binary compatibility
+    def this(supported: Set[ContentTypeRange]) = this(supported, None)
```

Comment:
do we want to mark these as deprecated as they're only here for binary compat?

Reply:
ok